packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -276,23 +277,30 @@ class ImageCache {
   /// [maximumSize] and [maximumSizeBytes].
   void _touch(Object key, _CachedImage image, TimelineTask? timelineTask) {
     assert(timelineTask != null);
+    assert(image is! _LiveImage);
```

Comment:
If we put a `LiveImage` in the `_cache`, it won't get disposed of properly.

Maybe they shouldn't be subclasses. But then we duplicate some logic.  I'm flexible on this one.